### DISCUSSION TITLE
SVGs with mask-* inline styles are stripped in qutebrowser

### DISCUSSION THREAD

**Discussion by xavierchanth** [qutebrowser/qutebrowser#8378] [Q&A] [answered]
I'm not sure if this is a bug or I need to change something in config (hence opened a discussion item first). I tried all the options I thought it might be, but certain styles are stripped from the html in qutebrowser.

This is what the svg looks like in chrome:

```xml
<svg class="gb-icon text-secondary m-2" style="mask-image: url(&quot;https://ka-p.fontawesome.com/releases/v6.6.0/svgs/regular/ban.svg?v=1&amp;token=a463935e93&quot;); mask-repeat: no-repeat; mask-position: center center; width: 24px; height: 24px;"></svg>
```

This is what that same svg looks like in qutebrowser:

```xml
<svg class="gb-icon text-secondary m-2" style="width: 24px; height: 24px;"></svg>
```

If you're curious, this is a webpage hosted with gitbook.
I can send a link if useful.

Also tested with `qutebrowser --temp-basedir`.

**Comment by The-Compiler** [comment, marked as answer]
Without having a way to reproduce, and without having your `:version` information, I can only guess... Yes, a link would be useful of course!

For starters, those `mask-*` properties [are only supported](https://caniuse.com/mdn-css_properties_mask-image) starting with Chromium 120, so you will need to be running QtWebEngine 6.8.

**Reply by xavierchanth** [reply]
I'm on 6.7!
Repro link: https://docs.noports.com/
Version info: https://bpa.st/W5VA

I'm on MacOS, and I see that there's still WIP for 6.8 support?
I can try building from source if you think it's worth a shot.

Thanks for your help!

**Reply by The-Compiler** [reply]
Yep, I can reproduce with Qt 6.7:

![image](https://github.com/user-attachments/assets/f91096f6-5d33-4a7c-a247-1f62fc25c07e)


But it looks just fine with Qt 6.8:

![image](https://github.com/user-attachments/assets/4208d3e6-b6f2-4822-b09c-2ff3719a5fa6)

Unfortunately there is no PyQt bundling Qt 6.8 out yet. Judging from [this answer](https://www.riverbankcomputing.com/pipermail/pyqt/2024-October/046045.html) and the [Qt 6.8 schedule](https://wiki.qt.io/Qt_6.8_Release), it might take another 10 days (at least) until that happens. I plan to release qutebrowser v3.4.0 bundling that shortly after.

In the meantime, you could experiment with [using `pyqt-bundle`](https://pyqt-builder.readthedocs.io/en/latest/pyqtbundle.html) to replace the Qt with a newer one (see #8242 for a rough guideline) and then run from sources with that, or perhaps try runninng `scripts/dev/build_release.py` to get an `.app`.

**Reply by xavierchanth** [reply]
Appreciate it, it's not make or break for me, but I will give it a shot and see how it goes.
